k8-operator/controllers/infisicalsecret_controller.go

```diff
@@ -154,9 +185,18 @@ func (r *InfisicalSecretReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}, nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
 func (r *InfisicalSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&secretsv1alpha1.InfisicalSecret{}).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
```

Comment:
Can we check if the operator has permission to watch secret resources?

Reply:
Yep, it goes the required permissions,
//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;**watch**;create;update;delete